add flags `default-pod-annotation` for e2e tests

erain opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:
1. If this is your first time, read our contributor guidelines https://git.k8s.io/community/contributors/guide#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
3. Follow the instructions for writing a release note: https://git.k8s.io/community/contributors/guide/release-notes.md
4. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

**What this PR does / why we need it**:

- New flag `default-pod-annotation ` in `TestContext` for e2e tests
- This flag represents a set of key=value pairs that describe annotations which are used for annotation-based features, e.g. seccomp profile, sandbox container etc.
- For example, This flag will be useful for testing sandbox containers like kata and gvisor.

**Release note**:
<!--  Write your release note:
1. Enter your extended release note in the below block. If the PR requires additional action from users switching to the new release, include the string "action required".
2. If no release note is required, just write "NONE".
-->
```release-note
- New flag `default-pod-annotation ` in `TestContext` for e2e tests
```

erain committed:
add flags `enable-sandbox-container` for e2e tests

- This flag will be useful for testing sandbox containers like kata and
gvisor.

Random-Liu commented on test/e2e/framework/test_context.go:
```diff
@@ -125,6 +125,9 @@ type TestContextType struct {
 	KubernetesAnywherePath string
 
+	// Indicates whether or not testing for sandbox containers.
+	EnableSandboxContainer bool
```
Let's make the name generic, something like `DefaultPodAnnotations map[string]string`.

Random-Liu replied:
And to justify this field, we can say this is useful to test existing annotation based features, e.g. default seccomp profile, sandbox container etc.

erain replied:
All done. Also, I updated the PR description for reflecting the changes we made. PTAL!

erain committed:
Make the flag for generic annotations:

- it accepts arbitrary annotations key-value pairs

Random-Liu commented:
LGTM. :)

So the flag usage will be `--default-pod-annotation=k1=v1,k2=v2`? Can you add a simple example in the flag description?

I think we are ready to send this upstream then. :)

erain committed:
Add flag usage for `default-pod-annotation`.

- Also remove the modified test case (all test cases changes will be
sent in a seperate PR.)

erain commented:
Thanks! Yes the usage you describe was correct, and the flag itself was implemented in https://github.com/kubernetes/apiserver/blob/master/pkg/util/flag/map_string_string.go . 

I've added the usage for the flag in the comment, and I removed the test case changes since that would come in a separate PR. Now sending this to upstream.
